src/features/diagnosticsProvider.ts

```diff
@@ -251,6 +251,12 @@ class DiagnosticsProvider extends AbstractSupport {
                     return;
                 }
 
+                // No problems published for virtual files
+                if(isVirtualCSharpDocument(document))
```

Comment:
Maybe call `shouldIgnoreDocument` instead. Is this a place where we should be mapping diagnostic locations?

Reply:
`shouldIgnoreDocument` has the following code:

```ts
 private shouldIgnoreDocument(document: TextDocument) {
        if (document.languageId !== 'csharp') {
            return true;
        }
        if (document.uri.scheme !== 'file' &&
            !isVirtualCSharpDocument(document)) {
            return true;
        }
        return false;
    }
```

So it would tell us that we should NOT ignore the virtual document.  

I'm not sure that mapping diagnostic locations is the correct response here because we already have a correctly mapped diagnostic for Index.cshtml in this case, so remapping the __virtual file would just lead to two identical diagnostics.

Reply:
whoops, I misread that one. sounds good